src/Hit/Issue.hs

```diff
@@ -121,6 +123,44 @@ createIssue title login = withOwnerRepo $ \token owner repo -> case token of
         errorMessage errorText
         pure $ Left $ ParseError errorText
 
+{- | Assign the user to the given 'Issue'.
+
+This function can fail assignment due to the following reasons:
+
+ * Auth token fetch failure
+ * Assignment query to GutHub failure
+
+The function should inform user about corresponding 'Error' in each case and
+continue working.
+-}
+assignIssue :: Issue -> Text -> IO ()
+assignIssue issue username = do
+    res <- withOwnerRepo $ \token owner repo -> case token of
+        Just auth -> do
+            let assignee :: Name User
+                assignee = makeName @User username
+            let curAssignees :: V.Vector (Name User)
+                curAssignees = V.map simpleUserLogin $ issueAssignees issue
+
+            if V.elem assignee curAssignees
+            then pure $ Right (issue, True)
+            else do
+                -- TODO: this is hack to cheat on GitHub library, as it
+                -- doesn't use the correct id in query.
+                let issId = mkIssueId (unIssueNumber $ issueNumber issue)
+                (, False) <<$>> GitHub.editIssue auth owner repo issId editOfIssue
+                    { editIssueAssignees = Just $ V.cons assignee curAssignees
+                    }
+
+        Nothing -> pure $ Left $ UserError "Can not get the GITHUB_TOKEN to assign you to the issue."
```

Comment:
I noticed that in `createIssue` a similar pattern-matching is used. Would it make sense to create a function like:

```haskell
withAuthOwnerRepo
    :: (Auth -> Name Owner -> Name Repo -> IO (Either Error a))
    -> IO (Either Error a)
```

which returns error on `Nothing` so we can use it in both `createIssue` and here? This can simplify code a little bit I guess, since we don't need to repeat this handling. I guess it's a good idea to move printing of the error message out from such functions, as you did, and print error on the side which calls the function. I think it makes sense that function that returns error shouldn't print it to avoid double printing.